Source/DafnyServer/boogie/DafnyProvider.cs

```diff
@@ -88,12 +88,15 @@ public DafnyModel(Model m, ViewOptions opts)
     /// </summary>
     private Model.Func MergeMapSelectFunctions(int arity) {
       int id = arity - 1;
-      // Need to make sure the name of the new function is unique
-      while (model.HasFunc("[" + ++id + "]")) { }
+      while (model.HasFunc("[" + id + "]")) {
+        id++; // increment id if the name "[" + id + "]" is already reserved
```

Comment:
This is not doing what the previous code did. Should perhaps line 90 be

    var id = arity;

Reply:
Actually, I'm now confused. If the name is already in use, do you need to create a different one? In fact, is it even correct to create a different one?

Reply:
The name shouldn't be in use but I added the check to prevent as situation in which there is a name clash due to some change to Dafny in the future

Reply:
My goal here is to give the function a meaningful name while avoiding name clashes with any any existing function

Reply:
You are quite right about changing what id is initialized to. Made this fix in my most recent commit